libstb/secvar/storage/secboot_tpm.c

```diff
@@ -370,8 +371,17 @@ static int secboot_tpm_store_init(void)
 
 	prlog(PR_DEBUG, "Initializing for pnor+tpm based platform\n");
 
-	// TODO: check physical presence here
-	// TODO: undefine indices if necessary here
+	/* Check if physical presence is asserted */
+	if (secvar_check_physical_presence()) {
+		prlog(PR_INFO, "Physical presence asserted, redefining NV indices, and resetting keystore\n");
+		rc = tss_nv_undefine_space(SECBOOT_TPMNV_VARS_INDEX);
+		if (rc)
+			goto error;
+
+		rc = tss_nv_undefine_space(SECBOOT_TPMNV_CONTROL_INDEX);
+		if (rc)
+			goto error;
+	}
```

Comment:
You have this comment in the behaviour - "This should cause them to be immediately redefined, causing a full reformat. " Hmm.. in this implementation PNOR is not getting formatted right away but will be in the next reboot. I wonder if we should just clear PNOR also here.

Reply:
This is before we even load in PNOR or the NV indices. In context of the code, we haven't checked yet if the indicies are even defined. In fact, there is an error here: we should probably ignore the return code, or filter out the "not defined" error.